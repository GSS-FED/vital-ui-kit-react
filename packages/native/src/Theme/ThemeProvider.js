// @flow
/* eslint-disable import/no-extraneous-dependencies */

import * as React from 'react';
import { ThemeProvider as Provider } from 'styled-components/native';
import defaultTheme from './theme';

export type Props = {
  theme?: any,
  children: React.ReactNode,
};

class ThemeProvider extends React.Component<Props> {
  state = {};
  render() {
    return (
      <Provider theme={{ ...defaultTheme, ...this.props.theme }}>
        {this.props.children}
      </Provider>
    );
  }
}

export default ThemeProvider;
