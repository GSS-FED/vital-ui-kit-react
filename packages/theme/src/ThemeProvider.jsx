// @flow

import React, { type Node } from 'react';
import { ThemeProvider as Provider } from 'styled-components';
import defaultVariables from './theme';
import defaultComponentsTheme from './theme/components';

type Props = {
  theme?: any,
  children: Node,
  componentTheme?: (props: defaultVariables) => any,
};

class ThemeProvider extends React.Component<Props> {
  static defaultProps = {
    theme: {},
    componentTheme: undefined,
  };

  render() {
    const { theme, componentTheme, children } = this.props;
    const combinedTheme = { ...defaultVariables, ...theme };
    const complieTheme = componentTheme || defaultComponentsTheme;
    const combinedWithComponentTheme = {
      ...combinedTheme,
      ...complieTheme(combinedTheme),
    };
    return (
      <Provider theme={combinedWithComponentTheme}>
        {children}
      </Provider>
    );
  }
}

export default ThemeProvider;
