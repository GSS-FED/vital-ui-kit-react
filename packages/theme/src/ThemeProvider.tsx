import React from 'react';
import { ThemeProvider as Provider } from 'styled-components';
import merge from 'lodash.merge';

import defaultVariables from './theme';
import defaultComponentsTheme from './theme/components';

type Props = {
  theme?: any;
  children: Node;
  componentTheme?: (props: typeof defaultVariables) => any;
};

class ThemeProvider extends React.Component<Props> {
  static defaultProps = {
    theme: {},
    componentTheme: (n: any) => n,
  };

  render() {
    const { theme, componentTheme, children } = this.props;

    const combinedTheme = { ...defaultVariables, ...theme };

    const combinedWithComponentTheme = merge(
      combinedTheme,
      defaultComponentsTheme(combinedTheme),
      componentTheme!(combinedTheme),
    );

    return (
      <Provider theme={combinedWithComponentTheme}>
        {children}
      </Provider>
    );
  }
}

export default ThemeProvider;
