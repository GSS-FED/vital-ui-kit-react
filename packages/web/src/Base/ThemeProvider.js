// @flow

import React, { type Node } from 'react';
import { ThemeProvider as Provider } from 'styled-components';
import defaultTheme from './styles/theme';
import IconProvider from '../Icon/components/IconProvider';
import defaultIcon from '../Icon/selection.json';

type Props = {
  theme?: {
    primaryList: string[],
    secondaryList: string[],
    blackList: string[],
  },
  icon?: any,
  children: Node
};

class ThemeProvider extends React.Component<Props> {
  static defaultProps = {
    theme: {},
    icon: {}
  };

  render() {
    const { theme, icon, children } = this.props;
    return (
      <Provider theme={{ ...defaultTheme, ...theme }}>
        <IconProvider icon={{ ...defaultIcon, ...icon }}>
          {children}
        </IconProvider>
      </Provider>
    );
  }
}

export default ThemeProvider;
