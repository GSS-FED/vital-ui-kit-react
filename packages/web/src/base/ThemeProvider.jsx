// @flow

import React, { type Node } from 'react';
import { ThemeProvider as Provider } from 'styled-components';
import defaultTheme from './theme';
import IconProvider from '../Icon/components/IconProvider';
import defaultIcon from '../Icon/selection.json';
import defeaultComponentsTheme from './theme/components';

type Props = {
  theme?: any,
  icon?: any,
  children: Node
};

class ThemeProvider extends React.Component<Props> {
  static defaultProps = {
    theme: {},
    icon: {}
  };

  render() {
    const { theme, icon, children } = this.props;
    const combinedTheme = { ...defaultTheme, ...theme }
    const combinedWithComponentTheme = {...combinedTheme, ...defeaultComponentsTheme(combinedTheme)};
    return (
      <Provider theme={combinedWithComponentTheme}>
        <IconProvider icon={{ ...defaultIcon, ...icon }}>
          {children}
        </IconProvider>
      </Provider>
    );
  }
}

export default ThemeProvider;
