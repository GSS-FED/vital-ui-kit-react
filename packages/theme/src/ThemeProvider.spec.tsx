import { combinedWithComponent, combineTheme } from './ThemeProvider';
import crmTheme from './theme/crm';
import defaultVariables from './theme';
import defaultComponentsThemeFn from './theme/components';
describe('ThemeProvider', () => {
  it('it should return default theme wihtout any input props', () => {
    let combineTheme = defaultVariables;
    const combinedWithComponentTheme = combinedWithComponent(
      combineTheme,
    );

    expect(combinedWithComponentTheme).toMatchSnapshot();
  });
});
