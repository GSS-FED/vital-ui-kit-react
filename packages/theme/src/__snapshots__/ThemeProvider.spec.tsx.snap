// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider it should return default theme wihtout any input props 1`] = `undefined`;
